### DISCUSSION TITLE
Add to symbols list ?

### DISCUSSION THREAD

**Discussion by idgraham** [zim-desktop-wiki/zim-desktop-wiki#2699] [Q&A] [answered]
Is it possible to add further symbols to what is offered default  (Insert > Symbols) ? A tick is an obvious candidate. 
Can this be done (a) from within the GUI ?   Or (b) only by getting into code ? 

If the latter, VERY clear instructions,please !

(I have run a search, and not found anything. )

Many thanks
Ian G,
Wales UK

**Comment by jaap-karssenberg** [comment]
Yes you can configure it, I believe the manual contains instructions on the
config

Op vr 22 nov. 2024 13:49 schreef idgraham ***@***.***>:

**Comment by jaap-karssenberg** [comment]
Manual page is here, but a bit short on details:
https://zim-wiki.org/manual/Plugins/Insert_Symbol.html

However if you go to the "insert symbol" dialog, there is a button "edit"
which opens up the relevant config file.

Hope this helps,

Jaap

Op vr 22 nov. 2024 17:16 schreef Jaap Karssenberg <
***@***.***>:

**Comment by idgraham** [comment]
Thanks for these prompt responses. I had tried the 'Edit' link, but then misunderstood 'application' in the following screen.
I'm still not understanding just what the 'Command' box requires, I'm afraid. Suppose I want to add a character with the Hex  U+ 221A and Decimal 8730, what should I put in that box ?

Many thanks

Ian G.

**Reply by Toggoren** [reply]
1. https://www.compart.com/en/unicode/U+221A  
  ![image](https://github.com/user-attachments/assets/51742773-7d11-4f12-a167-ad8553147391)

2. add ``\Sqrt	8730``  
  ![image](https://github.com/user-attachments/assets/941da3a1-cb59-4226-a995-eede769ada0d)

3. restart `zim`

result: 
![image](https://github.com/user-attachments/assets/138213e2-6408-45ec-9f60-c5f44a4971c3)

**Comment by jaap-karssenberg** [comment]
To open the config file, zim needs to know what text editor to use. This
command you are prompted for should be the name of the executable of your
texteditor - e.g. in my case it is set to "gedit". It is prompted only when
zim does not have a texteditor configured, later you can change it via the
preferences dialog on the "applications" tab.

Regards,

Jaap

**Reply by idgraham** [reply]
Thank you, both

**Reply by idgraham** [reply]
Well, happy to report I've been successful via direct edit of the symbols list. 
Which means the fact I haven't yet made the GUI edit (insert>symbol>edit) process work doesn't really matter - except that 

I am curious. I've juggled the various bits of data Toggoren posted, but keep getting the same error message : "could not find executable.....". I'll try and post a compo of some screen shots.
None of which takes away from the fact the Zim itself has very quickly become a central part of my tool-kit. Thanks, Jaap!
And finally, should anyone even less tech-savvy than me stumble on this thread: it does occur to me that the simplest work-around would be simply to create a page 'other symbols', and paste instances into that as they arise. 

![add compo](https://github.com/user-attachments/assets/60c39cba-51e5-4a4b-9e22-a86b120e309a)

**Reply by Toggoren** [reply]
Clicking on the `Edit` button tries to open a text file for editing in the default text editor.

If you are unlucky and the default editor was not found, you need to specify a few (two) “magic” lines in this window

![pasted_image](https://github.com/user-attachments/assets/6eac9a8c-cb7e-40f4-a6c7-8e7747e3ad4b)

Where `name` is a nice hint string.
And `command` is a string that can be inserted into a terminal emulator to start a text editor.

You must have some kind of text editor installed on your system. Unfortunately, I have never used the standard `notepad.exe`, so I can't say anything about its functionality.

[VSCodium](https://vscodium.com/) example

![pasted_image001](https://github.com/user-attachments/assets/ee63f971-a476-41a0-8633-ab7cde168216)


[Notepad++](https://notepad-plus-plus.org/downloads/) example

name: `Notepad++`
command: `"C:\\Program Files\\Notepad++\\notepad++.exe" -multiInst -notabbar -nosession -noPlugin`

Most Linux distributions already have a pretty good text editor, such as `kwrite` or `gedit`.



![image](https://github.com/user-attachments/assets/297b310e-4a26-4249-ab4a-1d64053dae58)

![pasted_image004](https://github.com/user-attachments/assets/6b8da2be-dcfd-4d6a-ab2d-fba9b38e8d53)

**Reply by idgraham** [reply, marked as answer]
Thank you !
